components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1438,11 +1493,33 @@ void RewardsServiceImpl::FetchGrants(const std::string& lang,
   if (!Connected()) {
     return;
   }
+#if !defined(OS_ANDROID)
+  bat_ledger_->FetchGrants(lang, payment_id, "", base::BindOnce(
+      &RewardsServiceImpl::OnFetchGrants,
+      AsWeakPtr()));
+#else
+  safetynet_check::ClientAttestationCallback attest_callback =
+      base::BindOnce(&RewardsServiceImpl::FetchGrantAttestationResult,
+          AsWeakPtr(), lang, payment_id);
+  safetynet_check_runner_.performSafetynetCheck("",
+      std::move(attest_callback));
+#endif
+}
 
-  bat_ledger_->FetchGrants(lang, payment_id, base::BindOnce(
+#if defined(OS_ANDROID)
+void RewardsServiceImpl::FetchGrantAttestationResult(const std::string& lang,
+    const std::string& payment_id,
+    bool result, const std::string& result_string) {
+  if (result) {
```

Comment:
Failure path(s) should occur before success path(s)

Reply:
If we are going that way, I would like to see a document that describes all that requirements for the code. Put it in the repo, put it in the wiki.

Reply:
NONCRITICAL
Covered by: brave/brave-browser#6235 (1)